File: src/types/template.ts
export interface Template {
  id: string
  user_id?: string | null
  name: string
  category?: string | null
  is_public: boolean
  preview_url?: string | null
  style_config: Record<string, unknown>
  usage_count: number
  created_at: string
  updated_at: string
}

File: backend/internal/domain/template.go
package domain

import (
	"encoding/json"
	"time"

	"github.com/google/uuid"
)

type Template struct {
	ID          uuid.UUID       `json:"id"`
	UserID      *uuid.UUID      `json:"user_id,omitempty"`
	Name        string          `json:"name"`
	Category    *string         `json:"category,omitempty"`
	IsPublic    bool            `json:"is_public"`
	PreviewURL  *string         `json:"preview_url,omitempty"`
	StyleConfig json.RawMessage `json:"style_config"`
	UsageCount  int             `json:"usage_count"`
	CreatedAt   time.Time       `json:"created_at"`
	UpdatedAt   time.Time       `json:"updated_at"`
}
